jerry-core/jrt/jrt-bit-fields.h

```diff
@@ -16,7 +17,29 @@
 #ifndef JERRY_BIT_FIELDS_H
 #define JERRY_BIT_FIELDS_H
 
-extern uint64_t __attr_const___ jrt_extract_bit_field (uint64_t, size_t, size_t);
-extern uint64_t __attr_const___ jrt_set_bit_field_value (uint64_t, uint64_t, size_t, size_t);
+/**
+ * Extract a bit-field.
+ *
+ * @param type type of container
+ * @param container container to extract bit-field from
+ * @param lsb least significant bit of the value to be extracted
+ * @param width width of the bit-field to be extracted
+ * @return bit-field's value
+ */
+#define JRT_EXTRACT_BIT_FIELD(type, container, lsb, width) \
+  (((container) >> lsb) & ((((type) 1) << (width)) - 1))
```

Comment:
I think the `do { ... } while (0)` trick could work here too. But if the compiler emit warnings, then it's Ok for me. The debug version of jerry runs really slow, so we should use assertions where they are necessary.

```C++
#define JRT_EXTRACT_BIT_FIELD(type, container, lsb, width) \
do \
{ \
  JERRY_ASSERT (lsb < JERRY_BITSINBYTE * sizeof (type)); \
  JERRY_ASSERT (width < JERRY_BITSINBYTE * sizeof (type)); \
  JERRY_ASSERT ((lsb + width) <= JERRY_BITSINBYTE * sizeof (type)); \
  (((container) >> lsb) & ((((type) 1) << (width)) - 1)) \
} while (0)
```